ako-gateway-api/nodes/dequeue_ingestion.go

```diff
@@ -215,16 +215,22 @@ func saveAviModel(modelName string, aviGraph *nodes.AviObjectGraph, key string)
 func (o *AviObjectGraph) ProcessRouteDeletion(key, parentNsName string, routeModel RouteModel, fullsync bool) {
 
 	parentNode := o.GetAviEvhVS()
+	routeTypeNsName := routeModel.GetType() + "/" + routeModel.GetNamespace() + "/" + routeModel.GetName()
 
-	found, childVSNames := akogatewayapiobjects.GatewayApiLister().GetRouteToChildVS(routeModel.GetType() + "/" + routeModel.GetNamespace() + "/" + routeModel.GetName())
+	found, childVSNamesTemp := akogatewayapiobjects.GatewayApiLister().GetRouteToChildVS(routeTypeNsName)
+	childVSNames := make([]string, len(childVSNamesTemp))
+
+	copy(childVSNames, childVSNamesTemp)
 	if found {
 		utils.AviLog.Infof("key: %s, msg: child VSes retrieved for deletion %v", key, childVSNames)
 
 		for _, childVSName := range childVSNames {
 			o.RemovePoolNameFromStringGroups(childVSName, parentNode, key)
-			nodes.RemoveEvhInModel(childVSName, parentNode, key)
+			removed := nodes.RemoveEvhInModel(childVSName, parentNode, key)
+			if removed {
```

Comment:
these 2 lines can be combined

Reply:
including a function call in an if statement will not look good